examples/nlp/language_modeling/megatron_gpt_ptune.py

```diff
@@ -1,135 +0,0 @@
-        )
-        register_taskdata_processor(processor_config.taskname, processor)
-
-    plugins = [NLPDDPPlugin()]
-    if cfg.trainer.precision == 16:
-        scaler = GradScaler(
-            init_scale=cfg.model.get('native_amp_init_scale', 2 ** 32),
-            growth_interval=cfg.model.get('native_amp_growth_interval', 1000),
-        )
-        plugins.append(NativeMixedPrecisionPlugin(precision=16, device='cuda', scaler=scaler))
-
-    if cfg.get('cluster_type', None) == 'BCP':
-        plugins.append(TorchElasticEnvironment())
-
-    trainer = Trainer(plugins=plugins, **cfg.trainer)
-
-    exp_manager(trainer, cfg.exp_manager)
-
-    # update resume from checkpoint found by exp_manager
-    resume_from_checkpoint = trainer._checkpoint_connector.resume_from_checkpoint_fit_path
-    logging.info(f'Resuming training from checkpoint: {resume_from_checkpoint}')
-
-    trainer._checkpoint_connector = CheckpointConnector(trainer, resume_from_checkpoint=resume_from_checkpoint)
-    # Override timer callback to a stateless one
-    for idx, callback in enumerate(trainer.callbacks):
-        if isinstance(callback, Timer):
-            trainer.callbacks[idx] = StatelessTimer(cfg.trainer.max_time,)
-
-    # hydra interpolation does not work here as the interpolation key is lost when PTL saves hparams
-    with open_dict(cfg):
-        cfg.model.precision = cfg.trainer.precision
-    model = MegatronGPTPTuneModel(cfg.model, trainer)
-    trainer.fit(model)
-
-    if cfg.model.data.test_ds.file_path:
-        logging.info("===========================================================================================")
-        logging.info("Starting the testing of the trained model on test set...")
-        trainer.test(model)
-        logging.info("Testing finished!")
-        logging.info("===========================================================================================")
-        # extract the path of the best checkpoint from the training, you may update it to any checkpoint
-        checkpoint_path = trainer.checkpoint_callback.best_model_path
-        tensor_parallel_size = cfg.model.tensor_model_parallel_size
-        pathobj = pathlib.Path(checkpoint_path)
-        checkpoint_folder = str(pathobj.parent)
-        checkpoint_name = str(pathobj.name)
-
-        rank = trainer.accelerator.training_type_plugin.local_rank
-        if tensor_parallel_size > 1:
-            # inject model parallel rank
-            checkpoint_path = os.path.join(checkpoint_folder, f'mp_rank_{rank:02d}', checkpoint_name)
-        else:
-            checkpoint_path = os.path.join(checkpoint_folder, checkpoint_name)
-
-        # Load the checkpoint
-        best_eval_model = MegatronGPTPTuneModel.load_from_checkpoint(
-            checkpoint_path=checkpoint_path, strict=False, trainer=trainer
-        )
-        logging.info(f'Best checkpoint path: {checkpoint_path}')
```

Comment:
Shall we add the logics of 

1. find the best checkpoint 
2.  convert model to nemo file

To the `megatron_gpt_prompt_tuning.py` file?

Reply:
Yeah, I think that will be good for external users pulling models from NGC. I'll add it.

Reply:
I will make sure converting to a nemo file is still showed in the notebook, btw

Reply:
If we already doing that in the `on_train_end`, it is not necessary. 
Finding best checkpoint and save to nemo is something generic enough for all the models. We need some utility function so every model can use it.

Reply:
Yes, it happens for all nemo models in `one_train_end`